doc/user/content/sql/create-materialized-view.md

```diff
@@ -68,6 +68,10 @@ indexes in each cluster you are referencing the materialized view in.
 this exists+add detail about using indexes to optimize materialized view
 stacking."
 
+### Limitations
```

Comment:
We shouldn't need to spell this out in the documentation as a limitation of views and materialized views, since there aren't that many scenarios where users would reference system objects in their object definition (and if they do, they shouldn't be using `SELECT *` to begin with :martazers:). If `SELECT *` and `NATURAL JOIN` against system objects are still allowed in ad-hoc querying, a release note + a warning under [`/sql/system-catalog/mz_catalog/`](https://materialize.com/docs/sql/system-catalog/mz_catalog/) should be enough.

This would be consistent with how we warn users that they can't create views that reference objects from `mz_internal`.

<img width="836" alt="Screenshot 2023-02-08 at 12 33 28" src="https://user-images.githubusercontent.com/23521087/217518900-dbfb325e-eed5-4f21-bb40-fe0d8c6f8c88.png">